lib/libzutil/zutil_nicenum.c

```diff
@@ -64,19 +65,22 @@ zfs_nicenum_format(uint64_t num, char *buf, size_t buflen,
 	uint64_t n = num;
 	int index = 0;
 	const char *u;
-	const char *units[3][7] = {
+	const char *units[6][7] = {
 	    [ZFS_NICENUM_1024] = {"", "K", "M", "G", "T", "P", "E"},
 	    [ZFS_NICENUM_BYTES] = {"B", "K", "M", "G", "T", "P", "E"},
-	    [ZFS_NICENUM_TIME] = {"ns", "us", "ms", "s", "?", "?", "?"}
+	    [ZFS_NICENUM_TIME] = {"ns", "us", "ms", "s", "?", "?", "?"},
+	    [ZFS_NICENUM_1000] = {"B", "K", "M", "G", "T", "P", "E"}
```

Comment:
Just thinking ahead - can you rename `ZFS_NICENUM_1000` to `ZFS_NICENUM_BYTES_1000`?  That way we could potentially add a SI base 1000 prefix for non-byte values, like:
```
	    [ZFS_NICENUM_1000] = {"", "K", "M", "G", "T", "P", "E"}
	    [ZFS_NICENUM_BYTES_1000] = {"B", "K", "M", "G", "T", "P", "E"}
 ```
I can imagine in the future that we make all non-byte values SI powers of 1000 by default.  It's kind of silly that we report zpool status error counters in powers of 1024, for example.

Reply:
The problem is the 5-char limit for nicenum values.  You'd give up precision with 2-char SI names.  Like, currently you can report "20.7M" but with the change you could only report "20Mi".  I'm sure we could fix the 5-char limit, but I don't know what kind of fallout there would be from it.  We've been reporting 5-char nicenum values for a long time...